Assets/Scripts/PowerUps/SpeedUpPower.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpeedUpPower : MonoBehaviour
+{
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    void OnTriggerEnter(Collider c) 
+    {
+        Debug.Log("XD");
+        Debug.Log(c);
+        if (c.CompareTag("Player"))
+        {
+            PlayerAnimation anim = c.GetComponent<PlayerAnimation>();
+
+            if (anim != null && !anim.hasSpeedPowerUp())
+            {
+                anim.GoFaster();
+                Destroy(gameObject);
+            }
```

Comment:
Consider creating a `PowerupManager` to handle all the states and state changes instead. This will not dirty `PlayerAnimation`, `PlayerMovement`, and will be more scalable when we support team powerups and etc.